contracts/core/abstract/ResultHandlerBase.sol

```diff
@@ -76,12 +76,25 @@ abstract contract ResultHandlerBase is IResultHandler, Initializable {
     // ============ Public Functions ============
 
     /// @inheritdoc IResultHandler
+    /// @dev This is left abstract as implementations need to decide how to handle the batch sender
+    /// @dev See postResultAndGetBatchSender for the core result posting logic
     function postResult(
         SedaDataTypes.Result calldata result,
         uint64 batchHeight,
         bytes32[] calldata proof
     ) public payable virtual override(IResultHandler) returns (bytes32);
 
+    /// @inheritdoc IResultHandler
+    function getResult(bytes32 requestId) public view override(IResultHandler) returns (SedaDataTypes.Result memory) {
+        SedaDataTypes.Result memory result = _resultHandlerStorage().results[requestId];
+        if (bytes(result.version).length == 0) {
+            revert ResultNotFound(requestId);
+        }
+        return _resultHandlerStorage().results[requestId];
```

Comment:
Can't we simply return the `result` variable? Or is this some Solidity shenanigans?

Reply:
YES! lol...not sure how I missed that!